src/praisonai-browser/praisonai_browser/cli/commands/browser.py

```
@@ -226,6 +226,40 @@ def clear_sessions(
+def _require_bridge(port: int = 8765) -> None:
+    """Pre-flight check: verify the bridge server is up with an extension connected.
+
+    Fails fast (a few seconds) with an actionable message instead of letting the
+    CLI hang for the full timeout on keepalive PING/PONG frames.
+    """
+    import json
+    import urllib.request
+
+    url = f"http://127.0.0.1:{port}/health"
+    try:
+        with urllib.request.urlopen(url, timeout=3) as resp:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)